pkg/mcp/sink/server_sink.go

```diff
@@ -0,0 +1,126 @@
+// Copyright 2018 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sink
+
+import (
+	"errors"
+	"io"
+	"sync/atomic"
+	"time"
+
+	"golang.org/x/time/rate"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/peer"
+	"google.golang.org/grpc/status"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/pkg/mcp/internal"
+	"istio.io/istio/pkg/mcp/monitoring"
+	"istio.io/istio/pkg/mcp/source"
+)
+
+var (
+	// For the purposes of logging rate limiting authz failures, this controls how
+	// many authz failus are logs as a burst every AUTHZ_FAILURE_LOG_FREQ.
+	authzFailureLogBurstSize = internal.EnvInt("AUTHZ_FAILURE_LOG_BURST_SIZE", 1)
```

Comment:
I think it's bad form to have an otherwise reentrant library pull in state from the ambient environment. I would prefer if these values came in as arguments when initializing this package, and let the wrapping command read the environment and supply the values.

That is, environment variables should handled like command-line arguments and read by the top-level command code.

Reply:
Fair enough. I've moved the envvar handling out of the common mcp packages and into the galley/configmap loading code. I've also added `Option` structs to the source/sink packages where callers can optionally set these sorts of overrides in addition to the other required parameters.